code/parse/sexp.cpp

```diff
@@ -18278,6 +18280,61 @@ void set_turret_secondary_ammo(ship_subsys *turret, int requested_bank, int requ
 	turret->weapons.secondary_bank_ammo[requested_bank] = requested_ammo;
 }
 
+// Goober5000
+int sexp_is_in_turret_fov(int node)
+{
+	char *target_ship_name;
+	char *turret_ship_name;
+	char *turret_subsys_name;
+	int target_shipnum, turret_shipnum, range;
+	object *target_objp, *turret_objp;
+	ship_subsys *turret_subsys;
+	vec3d tpos, tvec;
+
+	target_ship_name = CTEXT(node);
+	turret_ship_name = CTEXT(CDR(node));
+	turret_subsys_name = CTEXT(CDDR(node));
+	range = CDDDR(node) >= 0 ? eval_num(CDDDR(node)) : -1;
+
+	if (sexp_query_has_yet_to_arrive(target_ship_name) || sexp_query_has_yet_to_arrive(turret_ship_name))
+		return SEXP_CANT_EVAL;
+
+	// if ship is gone or departed, cannot ever evaluate properly.  Return NAN_FOREVER
+	if (mission_log_get_time(LOG_SHIP_DESTROYED, target_ship_name, NULL, NULL) || mission_log_get_time(LOG_SHIP_DEPARTED, target_ship_name, NULL, NULL) || mission_log_get_time(LOG_SELF_DESTRUCTED, target_ship_name, NULL, NULL)) {
+		return SEXP_NAN_FOREVER;
+	}
+	if (mission_log_get_time(LOG_SHIP_DESTROYED, turret_ship_name, NULL, NULL) || mission_log_get_time(LOG_SHIP_DEPARTED, turret_ship_name, NULL, NULL) || mission_log_get_time(LOG_SELF_DESTRUCTED, turret_ship_name, NULL, NULL)) {
+		return SEXP_NAN_FOREVER;
+	}
+
+	// find the two ships...
+	target_shipnum = ship_name_lookup(target_ship_name);
+	turret_shipnum = ship_name_lookup(turret_ship_name);
+	Assertion(target_shipnum >= 0, "Couldn't find target ship '%s' in sexp_is_in_turret_fov!", target_ship_name);
+	Assertion(turret_shipnum >= 0, "Couldn't find turreted ship '%s' in sexp_is_in_turret_fov!", turret_ship_name);
+
+	// ...and their objects
+	target_objp = &Objects[Ships[target_shipnum].objnum];
+	turret_objp = &Objects[Ships[turret_shipnum].objnum];
+
+	// find the turret
+	turret_subsys = ship_get_subsys(&Ships[turret_shipnum], turret_subsys_name);
+	Assertion(turret_subsys != nullptr, "Couldn't find turret subsystem '%s' on ship '%s' in sexp_is_in_turret_fov!", turret_subsys_name, turret_ship_name);
```

Comment:
Could it be possible that `turret_subsys_name` is invalid at this point due to a mission designer error?

Maybe if the ship class was changed from the original ship class. If that is possible then it would probably be better to use a `Warning` here or whatever is used in the other SEXPs.

Reply:
I wouldn't know, but that's an interesting contingency.  I'll change it to a `Warning`.